project/api/read.py

```diff
@@ -54,7 +54,8 @@ def get_profile(request, user):
         voted_solutions = Activity.objects.filter(
             account=a.id, civi__c_type="solution", activity_type__contains="pos"
         )
-        solution_threads = voted_solutions.distinct("thread__id").values_list(
+
+        solution_threads = voted_solutions.distinct().values_list(
```

Comment:
I think you are calling distinct before limiting the query to the `thread__id` column here, so the results might not be limited to distinct thread__ids.

Per this [SO](https://stackoverflow.com/questions/3852104/select-distinct-individual-columns-in-django), does something like `voted_solutions.values("thread__id").distinct()` work? It seems like that would appropriately remove duplicate thread__ids from the results.

Reply:
Let me give it a try

Reply:
It looks like it works. I added it to the PR and read up on the `values` method so I'm confident it serves the original purpose. The risk-averse side of me still wants to get it tested more thoroughly. Thanks for the feedback!